### PR TITLE
TW-1793: recent chats are not recent

### PR DESCRIPTION
**Pull request by sherlockvn**
###DESC:
- #1793 
- Change from member of recent groups to recents direct chat
### Demo:
- android:

https://github.com/linagora/twake-on-matrix/assets/43041967/7807d8f8-281c-4f8b-b5bc-6d98712924d2

- ios:

https://github.com/linagora/twake-on-matrix/assets/43041967/546a1eea-e14a-4585-baa9-2b407e201e35

- web:

https://github.com/linagora/twake-on-matrix/assets/43041967/05a55db1-daf4-495c-94cb-c0dc5e5770ed

### CODE REVIEW

File: lib/pages/search/recent_contacts_banner_widget.dart [hunk 1]
```diff
@@ -42,18 +45,21 @@ class PreSearchRecentContactsContainer extends StatelessWidget {
...
 
   @override
   Widget build(BuildContext context) {
+    final displayName = room.getLocalizedDisplayname(
+      MatrixLocals(L10n.of(context)!),
+    );
     return InkWell(
       onTap: () {
-        searchController.goToChatScreenFormRecentChat(user);
+        context.go('/rooms/${room.id}');
```
Comment (nqhhdev) [inline lib/pages/search/recent_contacts_banner_widget.dart:62]: IMO: Create a func in `controller`

File: lib/pages/search/recent_contacts_banner_widget.dart [hunk 2]
```diff
@@ -41,20 +43,19 @@ class PreSearchRecentContactsContainer extends StatelessWidget {
...
   @override
   Widget build(BuildContext context) {
+    final displayName = room.getLocalizedDisplayname(
+      MatrixLocals(L10n.of(context)!),
+    );
     return InkWell(
-      onTap: () {
-        searchController.goToChatScreenFormRecentChat(user);
-      },
+      onTap: () => context.go('/rooms/${room.id}'),
```
Comment (Te-Z) [inline lib/pages/search/recent_contacts_banner_widget.dart:58]: still here https://github.com/linagora/twake-on-matrix/pull/1800#discussion_r1615447655

Comment (sherlockvn) [reply lib/pages/search/recent_contacts_banner_widget.dart:58]: no need to depend on the controller, its will make the component harder to reuse.

Comment (Te-Z) [reply lib/pages/search/recent_contacts_banner_widget.dart:58]: alright then

**Review by Te-Z — APPROVED** [review]

**Review by nqhhdev — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**